patches/099-sqlite_api_3_26_0.patch

```diff
@@ -133,21 +462,650 @@ index d7477f57dbfca926fb8c8c78c63cb02a810e5c05..42b051da326d64d2f9fc920975a28316
    if (err != SQLITE_OK) {
      // Extended error codes cannot be enabled until a handle is
      // available, fetch manually.
+@@ -1726,30 +1739,6 @@ bool Connection::OpenInternal(const std::string& file_name,
+     return false;
+   }
+ 
+-  // TODO(shess): OS_WIN support?
+-#if defined(OS_POSIX) && !defined(OS_FUCHSIA)
+-  if (restrict_to_user_) {
+-    DCHECK_NE(file_name, std::string(":memory"));
+-    base::FilePath file_path(file_name);
+-    int mode = 0;
+-    // TODO(shess): Arguably, failure to retrieve and change
+-    // permissions should be fatal if the file exists.
+-    if (base::GetPosixFilePermissions(file_path, &mode)) {
+-      mode &= base::FILE_PERMISSION_USER_MASK;
+-      base::SetPosixFilePermissions(file_path, mode);
+-
+-      // SQLite sets the permissions on these files from the main
+-      // database on create.  Set them here in case they already exist
+-      // at this point.  Failure to set these permissions should not
+-      // be fatal unless the file doesn't exist.
+-      base::FilePath journal_path(file_name + FILE_PATH_LITERAL("-journal"));
+-      base::FilePath wal_path(file_name + FILE_PATH_LITERAL("-wal"));
+-      base::SetPosixFilePermissions(journal_path, mode);
+-      base::SetPosixFilePermissions(wal_path, mode);
+-    }
+-  }
+-#endif  // defined(OS_POSIX) && !defined(OS_FUCHSIA)
+-
+   // SQLite uses a lookaside buffer to improve performance of small mallocs.
+   // Chromium already depends on small mallocs being efficient, so we disable
+   // this to avoid the extra memory overhead.
+@@ -1821,16 +1810,8 @@ bool Connection::OpenInternal(const std::string& file_name,
+   const base::TimeDelta kBusyTimeout =
+     base::TimeDelta::FromSeconds(kBusyTimeoutSeconds);
+ 
+-  if (page_size_ != 0) {
+-    // Enforce SQLite restrictions on |page_size_|.
+-    DCHECK(!(page_size_ & (page_size_ - 1)))
+-        << " page_size_ " << page_size_ << " is not a power of two.";
+-    const int kSqliteMaxPageSize = 32768;  // from sqliteLimit.h
+-    DCHECK_LE(page_size_, kSqliteMaxPageSize);
+-    const std::string sql =
+-        base::StringPrintf("PRAGMA page_size=%d", page_size_);
+-    ignore_result(ExecuteWithTimeout(sql.c_str(), kBusyTimeout));
+-  }
++  const std::string sql = base::StringPrintf("PRAGMA page_size=%d", page_size_);
++  ignore_result(ExecuteWithTimeout(sql.c_str(), kBusyTimeout));
+ 
+   if (cache_size_ != 0) {
+     const std::string sql =
+diff --git a/sql/connection.h b/sql/connection.h
+index 8b9640785e98..7d936384ca65 100644
+--- a/sql/connection.h
++++ b/sql/connection.h
+@@ -36,7 +36,6 @@ class ProcessMemoryDump;
```

Comment:
There should also be a `#include "sql/internal_api_token.h"` change in this file.